### PR TITLE
More Refactoring

### PR DESCRIPTION
**Pull request by skipjack**
Tackling the refactoring mentioned in #2. This includes...

- Breaking out much of the render method into separate, more isolated components
- Using class methods instead of inline functions
- Investigating other blocks of code that could be simplified or re-worked
- Adding jsdoc comments for methods to make clear what each section is doing

Note that I'm not implementing or changing any behavior/design (e.g. #3), I'm simply cleaning things up a bit which should make adding features, redesigning, debugging, and maintaining the code base easier as we go forward. These changes should also make the codebase more approachable for other potential contributors.

Still have a little ways to go, will update when it's ready for review.

### CODE REVIEW

**Comment by skipjack** [comment]
@pksjce @NejcZdovc sorry this has taken a bit longer than I thought, but it's really close. I have a few more things to clean up but my goal is have this wrapped by the end of the week at the latest.

I requested your review as it's probably worth your both starting to look through and test even though there's a few things left. I did change a lot here and even once I finish I think there will still be much room for improvement.

**Comment by pksjce** [comment]
@skipjack - There seems to be a lot of work that has gone into this! 
I'm basically cloning this repo now. So will try it out now and give you any feedback

**Comment by skipjack** [comment]
@pksjce sounds good, thanks.

Yeah it turned into a bit more of a project than I intended it to but I'm pretty happy with the result and I hope you both are as well. I think it'll be easier to add more styling and feature with things cleaned up and modularized a bit.

After this, I'll be switching back to webpack.js.org work for the most part but I would like to come back here and contribute from time to time so I'm hoping we can keep it in a pretty clean, readable state so contributions are easy. Adding some tests and a simple CI process might be another nice thing at some point but probably not top priority at the moment.

**Comment by skipjack** [comment]
@sokra @pksjce @NejcZdovc this should be good to go. The main bits are now extracted into their own components, all component methods are documented, and all functionality should be working properly. I've been testing in "dev" mode the whole time... from my understanding it is the `name` prop (one the `Wrapper`) that triggers the switch between "dev" mode and "normal" mode.

In the current application that `name` prop was either `vote` or `todo` based on `page.url`. I left a note about this in `app.jsx` because if we continue with that behavior we'll have to remember to pass that back down when we re-incorporate it into the site.

Let me know if there's anything I missed that doesn't seem to be working properly.

File: src/components/wrapper/wrapper.jsx [hunk 1]
```diff
@@ -186,28 +78,30 @@ export default class Wrapper extends React.Component {
...
+                                    className={ `${block}__influence-section` } 
+                                    type="normal" />
+                                <Influence 
+                                    className={ `${block}__influence-section`} 
+                                    type="golden" />
                             </div>
 
-                            <div className="vote-app__influence-disclaimer">
+                            <div className={ `${block}__influence-disclaimer` }>
                                 DISCLAIMER: Since this feature is its Alpha stages, the formula for calculating influence may change.
```
Comment (pksjce) [inline src/components/wrapper/wrapper.jsx]: Are we still not finalised on the influence details?

Comment (skipjack) [inline src/components/wrapper/wrapper.jsx]: @pksjce good question, I'm not sure. I think @TheLarkInn or @sokra would know better, I'm assuming we probably we to get some more usage and feedback before removing it, however idk how much people are using it currently.

**Comment by pksjce** [comment]
@skipjack - I tested out the functionality. From the looks
1. User facing stuff seems the same. Didnt find any particular bug.
2. The admin interface needs improvement. Maybe create an issue for this?
3. What does refresh do? Does it reset your points to the initial point?

**Comment by skipjack** [comment]
> The admin interface needs improvement. Maybe create an issue for this?

Yeah, I'd say this is somewhat included in the redesign issue but not fully. Creating an issue with details so we can discuss a bit more sounds like a good idea.

> What does refresh do? Does it reset your points to the initial point?

You mean the refresh button beneath the user info? I believe it updates the points you have available. 

Now that I'm a bit more up to date on how this application works and it's all connected, I'd say in general the core functionality is there but the nuances/details like the "refresh" behavior, and even the admin behaviors like locking or archiving, are what needs a lot of work. The redesign discussed in #3 should help but there's likely a lot more that can be done to make the normal user, and admin, experience more straightforward and user-friendly.

**Pull request merged**